app/forms/candidate_interface/degree_wizard.rb

```diff
@@ -227,10 +227,14 @@ def persist!
       existing_degree = ApplicationQualification.find_by(id: id)
       if existing_degree.present?
```

Comment:
Suggested refactor:

```
      if existing_degree.present?
        existing_degree.update(attributes_for_persistence)
        clear_state!
      elsif all_attributes_for_persistence_present?
        ApplicationQualification.create!(attributes_for_persistence)
        clear_state!
      end
```
and then a private helper method:
```
      def all_attributes_for_persistence_present?
        attributes_for_persistence.slice(
          :qualification_type, :institution_name, :subject, :grade, :start_year, :award_year
        ).values.all?(&:present?)
      end
```

Reply:
thanks I've added